target/linux/kirkwood/files/arch/arm/boot/dts/kirkwood-c200-v1.dts

```diff
@@ -0,0 +1,378 @@
+		gpios = <&gpio1 2 GPIO_ACTIVE_HIGH>;
+	};
+
+	leds {
+		compatible = "gpio-leds";
+		pinctrl-0 = < &pmx_led_status_green &pmx_led_status_red
+				  &pmx_led_cloud_blue
+			      &pmx_led_disk1_green &pmx_led_disk1_red
+			      &pmx_led_disk2_green &pmx_led_disk2_red
+			      &pmx_led_disk_fail_green &pmx_led_disk_fail_red
+			      &pmx_led_usb1_green &pmx_led_usb1_red
+			      &pmx_led_usb2_green &pmx_led_usb2_red
+			      >;
+		pinctrl-names = "default";
+
+		led_status_green: status_green {
+			label = "green:status";
+			gpios = <&gpio1 7 GPIO_ACTIVE_LOW>;
+		};
+
+		led_status_red: status_red {
+			label = "red:status";
+			gpios = <&gpio1 6 GPIO_ACTIVE_LOW>;
+		};
+
+		cloud_blue {
+			label = "blue:cloud";
+			gpios = <&gpio1 8 GPIO_ACTIVE_LOW>;
+		};
+
+		disk1_green {
+			label = "green:disk1";
+			gpios = <&gpio0 15 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "ata1";
+		};
+
+		disk1_red {
+			label = "red:disk1";
+			gpios = <&gpio0 17 GPIO_ACTIVE_LOW>;
+		};
+
+		disk2_green {
+			label = "green:disk2";
+			gpios = <&gpio0 16 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "ata2";
+		};
+
+		disk2_red {
+			label = "red:disk2";
+			gpios = <&gpio0 14 GPIO_ACTIVE_LOW>;
+		};
+
+		disk_fail_green {
+			label = "green:disk-fail";
+			gpios = <&gpio1 11 GPIO_ACTIVE_LOW>;
+		};
+
+		disk_fail_red {
+			label = "red:disk-fail";
+			gpios = <&gpio1 10 GPIO_ACTIVE_LOW>;
+		};
+
+		usb1_green {
+			label = "green:usb1";
+			gpios = <&gpio1 13 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "usbport";
+			trigger-sources = <&hub_port2>;
+		};
+
+		usb1_red {
+			label = "red:usb1";
+			gpios = <&gpio1 12 GPIO_ACTIVE_LOW>;
+		};
+
+		usb2_green {
+			label = "green:usb2";
+			gpios = <&gpio1 15 GPIO_ACTIVE_LOW>;
+			linux,default-trigger = "usbport";
+			trigger-sources = <&hub_port1>;
+		};
+
+		usb2_red {
+			label = "red:usb2";
+			gpios = <&gpio1 14 GPIO_ACTIVE_LOW>;
+		};
+	};
+};
+
+&eth0 {
+	status = "okay";
+	ethernet0-port@0 {
+		phy-handle = <&ethphy9>;
+	};
+};
+
+&i2c0 {
+	status = "okay";
+
+	rtc@30 {
+		compatible = "s35390a";
+		reg = <0x30>;
+	};
+
+	lm63@4c {
+		compatible = "national,lm63";
+		reg = <0x4c>;
+	};
+};
+
+&mdio {
+	status = "okay";
+	ethphy9: ethernet-phy@9 {
+		reg = <9>;
+	};
+};
+
+&nand {
+	status = "okay";
+	chip-delay = <40>;
+
+	partition@0 {
+		label = "uboot";
+		reg = <0x0000000 0x200000>;
+	};
+	partition@200000 {
+		label = "certificate";
+		reg = <0x0200000 0x100000>;
+	};
+	partition@300000 {
+		label = "preset_cfg";
+		reg = <0x0300000 0x100000>;
+	};
+	partition@400000 {
+		label = "dev_params";
+		reg = <0x0400000 0x100000>;
+	};
+	partition@500000 {
+		label = "active_bank";
+		reg = <0x0500000 0x0100000>;
+	};
+	partition@600000 {
+		label = "magic";
+		reg = <0x0600000 0x0100000>;
+	};
+	partition@700000 {
+		label = "bank1";
+		reg = <0x0700000 0x2800000>;
+	};
+	partition@2f00000 {
+		label = "bank2";
+		reg = <0x2f00000 0x2800000>;
+	};
+	partition@5a00000 {
```

Comment:
What's between 0x5700000 and 0x5a00000?

Reply:
It's extra space. Factory partitions are mess:

```
Creating 10 MTD partitions on "orion_nand":
0x000000700000-0x000002f00000 : "romdisk"
0x000000000000-0x000000200000 : "u-boot"
0x000000200000-0x000000300000 : "certificate"
0x000000300000-0x000000400000 : "preset_cfg"
0x000000400000-0x000000500000 : "dev_params"
0x000000500000-0x000000600000 : "active_bank"
0x000000600000-0x000000700000 : "magic"
0x000000700000-0x000002f00000 : "bank1"
0x000002f00000-0x000005700000 : "bank2"
Setup romdisk MTD parition (0x0000031f7440-0x000005700000)
0x000005a00000-0x000007a00000 : "ubi"
```

I use factory partitions to keep revert to stock.

Reply:
If you don't put a partition there and leave it empty be intention, please put a comment in the DTS.

E.g.
https://github.com/openwrt/openwrt/blob/master/target/linux/ath79/dts/qca9561_tplink_eap225-wall-v2.dts#L106
or
https://github.com/openwrt/openwrt/blob/master/target/linux/ath79/dts/qca9563_tplink_tl-wpa8630p-v2.1-eu.dts#L33

Reply:
Done.